### PR TITLE
Store the IP firewall info in Redis

### PR DESCRIPTION
**Pull request by thedae**
From: https://app.clubhouse.io/cartoteam/story/98420/store-the-ip-firewall-information-in-redis

Also please check this one-liner to populate Redis with the information that it's already in database via Rails console:

```
Carto::User.find_each do |user|
  if user.dbdirect_effective_ips.any?
    $users_metadata.HSETNX('dbdirect:' + user.username, 'ips', user.dbdirect_effective_ips.join(','))
  end
end
```

### CODE REVIEW

File: config/app_config.yml.sample [hunk 1]
```diff
@@ -789,13 +789,10 @@ defaults: &defaults
...
-      enabled: false
-      project_id: ''
-      rule_name: ''
-      network: ''
-      target_tag: ''
-      ports: 'tcp:5432'
+    metadata_persist:
+      enabled: true
+      prefix_namespace: 'dbdirect:'
+      hash_key: 'ips'
```
Comment (ilbambino) [inline config/app_config.yml.sample:795]: do you think that it makes sense to have the key of the key as a config value? Is it something that would need to be changed?

**Review by ilbambino — COMMENTED** [review]
I am no expert in Rails, but it looks good to me. 
I like the PRs where there is more red than green 😄 

There is a rake to check if DB and Redis are in sync, but there is nothing to sync them. How are we planning to do that?

File: config/app_config.yml.sample (hunk 1 above)
Comment (thedae) [reply config/app_config.yml.sample:795]: About the initial sync: I left a comment in this PR with a one-liner to be executed in Rails console once this is deployed to staging/prod. Let me add it again:
```
Carto::User.find_each{ |user| $users_metadata.HSETNX('dbdirect:' + user.username, 'ips', user.dbdirect_effective_ips.join(',')) }
```
Regarding the config file, I doubt those keys are something that are going to be changed, but to be honest, I personally don't like to add "magic strings" directly in the code, not even as class constants. Having said that, I'm open for modifications if you think it's not a good idea :)

Comment (ilbambino) [reply config/app_config.yml.sample:795]: 🙈 I am blind sorry

I don't like magic strings either. But as it is something that would need to be changed also in `pg-proxy` (same comment could then apply to `prefix_namespace`. 
I don't have strong opinion, so LGTM

**Review by ilbambino — APPROVED** [review]

**Pull request merged**